### ISSUE TITLE
Invalid privacy manifest

### ISSUE THREAD

**Issue by denhams** [mapbox/mapbox-common-ios#245]
MapboxCommon 23.9.1
MapboxCoreMaps 10.16.6
MapboxMap 10.16.6

When submitting my app to TestFlight, I received an email from Apple:

_ITMS-91056: Invalid privacy manifest - The PrivacyInfo.xcprivacy file from the following path is invalid: “Frameworks/MapboxCommon.framework/PrivacyInfo.xcprivacy”. While no action is required at this time, starting May 1, 2024, when you upload a new app or app update, keys and values in your app’s privacy manifest must be in a valid format. For more details about privacy manifest files, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files._

**Comment by tattn** [comment]
I'm also experiencing the same issue.

Upon checking the contents of the file, I noticed that the value for keys such as `NSPrivacyAccessedAPITypeReasons` within `MapboxCommon.framework/PrivacyInfo.xcprivacy` is not in an array format.

For instance, the following:
```
<dict>
	<key>NSPrivacyAccessedAPITypeReasons</key>
	<string>CA92.1</string>
	<key>NSPrivacyAccessedAPIType</key>
	<string>NSPrivacyAccessedAPICategoryUserDefaults</string>
</dict>

```

Should be like this:
```
<dict>
	<key>NSPrivacyAccessedAPITypeReasons</key>
	<array>
		<string>CA92.1</string>
	</array>
	<key>NSPrivacyAccessedAPIType</key>
	<string>NSPrivacyAccessedAPICategoryUserDefaults</string>
</dict>

```

**Comment by tattn** [comment]
Related issue: https://github.com/mapbox/mapbox-maps-ios/issues/2145

**Comment by DorkMatter** [comment]
Fixed in v24.3.0
